tsl/test/expected/delete_exclusion.out

```diff
@@ -0,0 +1,476 @@
+ Custom Scan (HypertableModify) (actual rows=0 loops=1)
+   ->  Delete on metrics_date (actual rows=0 loops=1)
+         Delete on metrics_date metrics_date_1
+         Delete on _hyper_4_4_chunk metrics_date_2
+         ->  Append (actual rows=1 loops=1)
+               ->  Seq Scan on metrics_date metrics_date_1 (actual rows=0 loops=1)
+                     Filter: ("time" = '01-01-2000'::date)
+               ->  Index Scan using _hyper_4_4_chunk_metrics_date_time_idx on _hyper_4_4_chunk metrics_date_2 (actual rows=1 loops=1)
+                     Index Cond: ("time" = '01-01-2000'::date)
+(9 rows)
+
+:PREFIX DELETE FROM metrics_timestamp WHERE time = '2000-01-01';
+                                                                   QUERY PLAN                                                                   
+------------------------------------------------------------------------------------------------------------------------------------------------
+ Custom Scan (HypertableModify) (actual rows=0 loops=1)
+   ->  Delete on metrics_timestamp (actual rows=0 loops=1)
+         Delete on metrics_timestamp metrics_timestamp_1
+         Delete on _hyper_5_5_chunk metrics_timestamp_2
+         ->  Append (actual rows=1 loops=1)
+               ->  Seq Scan on metrics_timestamp metrics_timestamp_1 (actual rows=0 loops=1)
+                     Filter: ("time" = 'Sat Jan 01 00:00:00 2000'::timestamp without time zone)
+               ->  Index Scan using _hyper_5_5_chunk_metrics_timestamp_time_idx on _hyper_5_5_chunk metrics_timestamp_2 (actual rows=1 loops=1)
+                     Index Cond: ("time" = 'Sat Jan 01 00:00:00 2000'::timestamp without time zone)
+(9 rows)
+
+:PREFIX DELETE FROM metrics_timestamptz WHERE time = '2000-01-01';
+                                                                     QUERY PLAN                                                                     
+----------------------------------------------------------------------------------------------------------------------------------------------------
+ Custom Scan (HypertableModify) (actual rows=0 loops=1)
+   ->  Delete on metrics_timestamptz (actual rows=0 loops=1)
+         Delete on metrics_timestamptz metrics_timestamptz_1
+         Delete on _hyper_6_6_chunk metrics_timestamptz_2
+         ->  Append (actual rows=1 loops=1)
+               ->  Seq Scan on metrics_timestamptz metrics_timestamptz_1 (actual rows=0 loops=1)
+                     Filter: ("time" = 'Sat Jan 01 00:00:00 2000 PST'::timestamp with time zone)
+               ->  Index Scan using _hyper_6_6_chunk_metrics_timestamptz_time_idx on _hyper_6_6_chunk metrics_timestamptz_2 (actual rows=1 loops=1)
+                     Index Cond: ("time" = 'Sat Jan 01 00:00:00 2000 PST'::timestamp with time zone)
+(9 rows)
+
+ROLLBACK;
+-- stable constraints
+-- should have ChunkAppend since constraint is stable
+-- should only hit 1 chunk and base table
```

Comment:
For stable constraints, if we prepare a statement and it uses a generic plan, is it going to work correctly? In this case, the stable functions should be re-evaluated and chunk exclusion should be done again.

Reply:
For stable constraints this happens at executor startup not during planning, so every execution will reevaluate constraints.

Reply:
Should we need to add some tests here?